tools/rtpsrelay/Config.cpp

```diff
@@ -0,0 +1,287 @@
+  });
+  cached_restart_detection_.default_if_empty([&](bool default_val) {
+    restart_detection(default_val);
+  });
+  cached_admission_control_queue_size_.default_if_empty([&](size_t default_val) {
+    admission_control_queue_size(default_val);
+  });
+  cached_admission_control_queue_duration_.default_if_empty([&](const TimeDuration& default_val) {
+    admission_control_queue_duration(default_val);
+  });
+  cached_max_ips_per_client_.default_if_empty([&](size_t default_val) {
+    max_ips_per_client(default_val);
+  });
+  cached_rejected_address_duration_.default_if_empty([&](const TimeDuration& default_val) {
+    rejected_address_duration(default_val);
+  });
+  cached_admission_max_participants_high_water_.default_if_empty([&](size_t default_val) {
+    admission_max_participants_high_water(default_val);
+  });
+  cached_admission_max_participants_low_water_.default_if_empty([&](size_t default_val) {
+    admission_max_participants_low_water(default_val);
+  });
+}
+
+void Config::on_data_available(InternalDataReader_rch reader)
+{
+  OpenDDS::DCPS::ConfigReader::SampleSequence samples;
+  OpenDDS::DCPS::InternalSampleInfoSequence infos;
+  reader->read(samples, infos, DDS::LENGTH_UNLIMITED,
+               DDS::NOT_READ_SAMPLE_STATE, DDS::ANY_VIEW_STATE, DDS::ANY_INSTANCE_STATE);
+  for (size_t idx = 0; idx != samples.size(); ++idx) {
+    const auto& info = infos[idx];
+    const auto& pair = samples[idx];
+    if (info.valid_data) {
+      if (pair.key() == RTPS_RELAY_LIFESPAN) {
+        cached_lifespan_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_INACTIVE_PERIOD) {
+        cached_inactive_period_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_WARNINGS) {
+        cached_log_warnings_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_DISCOVERY) {
+        cached_log_discovery_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_ACTIVITY) {
+        cached_log_activity_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_HTTP) {
+        cached_log_http_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_THREAD_STATUS) {
+        cached_log_thread_status_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_THREAD_STATUS_SAFETY_FACTOR) {
+        cached_thread_status_safety_factor_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_UTILIZATION_LIMIT) {
+        cached_utilization_limit_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_UTILIZATION_CHANGES) {
+        cached_log_utilization_changes_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_RELAY_STATISTICS) {
+        cached_log_relay_statistics_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_HANDLER_STATISTICS) {
+        cached_log_handler_statistics_.set(pair.value());
+      } else if (pair.key() == RTPS_RELAY_LOG_PARTICIPANT_STATISTICS) {
+        cached_log_participant_statistics_.set(pair.value());
```

Comment:
I think this and the RTPS_RELAY_PUBLISH_PARTICIPANT_STATISTICS key below are related to participant statistics which removed in PR #5109